docs/apis/plugintypes/theme/images.md

```diff
@@ -0,0 +1,90 @@
+
+The following code snippet illustrates how to make use of your images within your layout file so they can be inserted by your layout template.
+
+```php title="theme/yourtheme/layout/somelayout.php"
+$templatecontext = [
+...
+$imageone => $OUTPUT->pix_url('imageone', 'theme'),
+$imagetwo => $OUTPUT->pix_url('subdir/imagetwo', 'theme'),
+...
+];
+
+echo $OUTPUT->render_from_template('theme_yourtheme/somelayout', $templatecontext);
+```
+
+A method of Moodle's output library is utilized to generate the URL to the image. It's not too important how that function works, but it is important that it's used, as it's what allows images within Moodle to be overridden.
+**DO NOT** include the image file extension. Moodle will work it out automatically and it will not work if you do include it.
+
+```handlebars title="theme/yourtheme/templates/somelayout.mustache"
+<img src="{{{imageone}}}" alt="Please give your image alt text or set the role to presentation" width="50" height="50">
+<img src="{{{imagetwo}}}" alt="Please give your image alt text or set the role to presentation" width="50" height="50">
+```
+
+The following is how you would use the images from within CSS, SCSS or LESS as background images.
+
+```css
+.divone {
+    background-image: url([[pix:theme|imageone]]);
+}
+
+.divtwo {
+    background-image: url([[pix:theme|subdir/imagetwo]]);
+}
+```
+
+In this case, special notations that Moodle looks for must be used. Whenever Moodle hands out a CSS file it first searches for all `something` tags and replaces them with what is required.
+
+The final thing to notice with both of the cases above is that at no point is the image file extension included.
+The reason for this leads us into the next topic, how to override images.
+
+From within a theme you can VERY easily override any standard image within Moodle by simply adding the replacement image to the theme's pix directory in the same sub directory structure as it is in Moodle.
+So, for instance, if there is a need to override the following images:
+
+1. `/pix/moodlelogo.png`
+1. `/pix/i/user.svg`
+1. `/mod/chat/pix/monologo.svg`
+
+Simply add the replacement images to the theme in the following locations:
+
+1. `/theme/themename/pix_core/moodlelogo.png`
+1. `/theme/themename/pix_core/i/user.svg`
+1. `/theme/themename/pix_plugins/mod/chat/monologo.svg`
+
+:::note
+
+A `pix_core` directory has been created in the theme to store the replacement images. For a specific activity module like chat, the directory `pix_plugins/mod/chat` is needed. This directory is `pix_plugins` and then the plugin type (mod) and then the plugin name (chat).
+
+:::
+
+Now the other very cool thing to mention is that Moodle looks for not just replacements of the same image type (svg, png, jpg, gif, etc...) but also replacements in any image format. That's why when working with our images above, the image file extension was never specified.
```

Comment:
We've mostly been steering clear of overly 'friendly' workding like 'very cool'.